pkg/volume/plugins.go

```diff
@@ -42,6 +42,13 @@ type VolumeOptions struct {
 	RootContext string
 }
 
+// System is an interface that abstracts certain operations on a host operating
+// system that a volume might want to execute.
+type System interface {
```

Comment:
clearer: yes, I think you don't need a system object, you need to add either one method `WriteFile` or to remove the mount.Interface that is passed around and add two methods `WriteFile` and `GetMounter`.  The latter is probably cleaner overall, but more work.